bokeh/models/widgets/inputs.py

```diff
@@ -208,6 +217,73 @@ class DatePicker(InputWidget):
     A callback to run in the browser whenever the current date value changes.
     """)
 
+
+class ColorPicker(InputWidget):
+    ''' Color picker widget
+
+    '''
+
+    color = Color(default='#000000', help="""
+    The initial color of the picked color (named or hexadecimal)
+    """)
+
+    callback = Instance(Callback, help="""
+    A callback to run in the browser whenever the current date value changes.
+    """)
+
+    def __init__(self, **kwargs):
+        if 'color' in kwargs:
+            potential_color = kwargs['color']
+            color = None
+
+            if isinstance(potential_color, str):
+                if re.search(r'^#(?:[0-9a-fA-F]{3}){1,2}$', potential_color):  # hex
+                    color = potential_color.upper()
+                elif hasattr(named, potential_color.lower()):  # named
+                    color = getattr(named, potential_color.lower()).to_hex()
+            elif isinstance(potential_color, HSL):  # hsl
+                color = potential_color.to_rgb().to_hex()
+            elif isinstance(potential_color, RGB):  # rgb
+                color = potential_color.to_hex()
+            elif isinstance(potential_color, tuple) and len(potential_color) == 3:  # tuple
+                color = RGB(*potential_color).to_hex()
+
+            if color is None:
+                raise ValueError("expected and element of either String, HSL, RGB or Tuple(r,g,b). " + \
+                                 "String must be a valid hexadecimal color or a named color from this list: " + \
+                                 f"{list(NamedColor)}. got {potential_color}")
+            else:
+                kwargs['color'] = color
+        super(ColorPicker, self).__init__(**kwargs)
```

Comment:
@xavArtley what's preventing from just letting the `Color` property handle all this validation? I would expect not needing to have an `__init__` at all.

Reply:
Actually in typescript I don't know how to handle Color other than hexadecimal string

Reply:
I see. There's probably a way we can handle things at the property level, such that serialization always happens to RGB hex values. Let me think on it.